### PR TITLE
Allow building pkg for Mac App Store

### LINKED ISSUE #1056: pkg can be published to Mac App Store

### PR DESCRIPTION
**Pull request by jperedadnr**
### Issue

Fixes #1056 
Requires a follow-up PR in GluonFX Maven/Gradle plugins, to implement ReleaseConfiguration changes.

### Progress

- [ ] Change must not contain extraneous whitespace
- [ ] License header year is updated, if required
- [ ] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

**Review by johanvos — APPROVED** [review]

File: src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java [hunk 1]
```diff
@@ -58,24 +69,64 @@
     private static final String ERRLINK = "Please check https://docs.gluonhq.com/ for more information.";
 
     private static final String KEYCHAIN_ERROR_MESSAGE = "errSecInternalComponent";
+    private static final List<String> SANDBOX_KEYS = List.of(
+            "com.apple.security.app-sandbox",
+//            "com.apple.security.cs.allow-jit", "com.apple.security.cs.allow-dyld-environment-variables",
```
Comment (erwin1) [inline src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java:74]: get rid of these?

File: src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java [hunk 2]
```diff
@@ -188,17 +239,40 @@ private boolean verifyCodesign(Path target) throws IOException, InterruptedExcep
...
+            if (provisionProfile != null) {
+                NSDictionaryEx profileEntitlements = provisionProfile.getEntitlements();
+                Arrays.stream(profileEntitlements.getAllKeys())
+                        .filter(key -> dict.get(key) == null)
+                        .forEach(key -> dict.put(key, profileEntitlements.get(key)));
+            }
+
+            dict.saveAsXML(tmpEntitlements);
+        } catch (ParserConfigurationException | ParseException | SAXException | PropertyListFormatException e) {
+            e.printStackTrace();
```
Comment (erwin1) [inline src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java:266]: probably don't ignore exceptions here. you want to fail the build of this happens?

File: src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java (hunk 1 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java:74]: Makes sense. Those are used by the regular jpackage for the JVM, but native-image shouldn't need them

File: src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java (hunk 2 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/macos/CodeSigning.java:266]: Makes sense, since the user can add a malformed plist file.

**Review by erwin1 — APPROVED** [review]

**Pull request merged**